### ISSUE TITLE
[Bug] Ads in app

### ISSUE THREAD

**Issue by Exit24** [whoeevee/EeveeSpotify#422]
### Is this issue appropriate?

- [X] I have read the document in its entirety

### Is this issue unique to EeveeSpotify?

- [X] I was unable to reproduce my issue using the stock Spotify app

### Have you searched the existing issues?

- [X] This is a unique issue

### Describe the bug.

Is anyone else getting an advertisement on the main screen in the app now? I can easily close it by hitting the X button, but it comes back every time I open the app. 

Example:
https://imgur.com/a/n96W4HO

**Comment by whoeevee** [comment]
i believe closing the ad after app launch is not a big problem
you can also try enabling overwrite configuration

**Comment by ChipSkylark37** [comment]
Isn't the point of this application not to have Ads? Amongst other features of course. 

Yes it's not a big issue if you can X out of it, but how can we help to diagnose /why/ they are there and maybe we can figure out how to block them better as there's a number of other 'closed' issues reporting the same thing. 

Simply closing the issue doesnt seem right.

**Comment by whoeevee** [comment]
yeah, the main point of EeveeSpotify is to make the app think you have a premium subscription and unlock some features like shuffle, free listening, etc. it also disables ads — you can listen to tracks freely. however, some users may see an ads banner on the homepage occasionally. this banner is easily closable, and I believe it’s not something that really needs to be fixed or causes much inconvenience. you can try enabling overwrite configuration option, which may help as it replaces the remote configuration from server (containing many UI/UX parameters) with the dumped premium one

however, to fully resolve this issue, i would need to investigate spotify ad logic, see exactly how the app determines to show ads, etc., which i’m not currently doing. i don’t even get these banners myself — definitely not on app launch. might see one once every few months, which I just close

i think I’ll leave the issue open; maybe someday someone will open a pull request with a fix for these ads if interested. i hope you understand my point of view regarding this

**Comment by ChipSkylark37** [comment]
Understood - I can imagine its very difficult to troubleshoot a 'bug' that only displays every few months, or more frequently for others, and sporadically at that. For

Is there anything we can do to help diagnose? I don't have any real coding experience, but if we can replicate it and see what IP it's reaching out to (maybe a specific 'ad server') that can help to block it? 

Personally, I run Pi-Hole at home which would log what IPs the device is reaching out to.

**Comment by asdfzxcvbn** [comment]
eevee is right. all eeveespotify does is make spotify think you have premium, and you can tell it works with features like unlimited skips and such. a minor banner ad that doesn't even show up for him often isn't too big of a deal, simply close the ad

**Comment by whoeevee** [comment]
some updates: recently i started constantly getting ads on the home page, so looked into it. turned out that spotify has very complicated ads logic, most of which is disabled if the `ads` value is false in product state. ads on the home screen were still shown and somewhat broken due to layout, which is received dynamically from the server. therefore, I need to remove them in a similar way to how I enabled liked songs. however, the ads are gone now. I’ll probably implement this solution as soon as they start appearing again

**Comment by ChipSkylark37** [comment]
I've also recently (last night) had ads playing in between songs when casting to a Google Home Max. 

I know sometimes when casting it does a "handoff" to that device and that device streams direct from the service which could introduce ads. I'm unsure if that's the case or not.

**Comment by asdfzxcvbn** [comment]
basically, the ad is delivered to the client via a protobuf response. the client sets up the UI for the ad (`UICollectionView`) and there's no _effective_ way to hide it. it's all implemented in swift too, which is known for being extremely difficult if not downright impossible to hook. a theoretical solution would be deserializing the protobuf response, modifying it, then replacing the response. however it's too much effort and likely won't be implemented any time soon. will leave open for now

**Comment by ChipSkylark37** [comment]
Forgive me because this is way over my head, but just curious if it is even remotely possible:

- The 'Hide Distracting Content' option inside of Safari works to hide certain elements on a persistent basis for web pages. Nothing like this could be used inside Spotify, right? Assuming that works entirely different than this.

**Comment by Richard-NDC** [comment]
i have a premium account so maybe your team can fix it?

**Comment by Sam00github** [comment]
> ### Is this issue appropriate?
> * [x]  I have read the document in its entirety
> 
> ### Is this issue unique to EeveeSpotify?
> * [x]  I was unable to reproduce my issue using the stock Spotify app
> 
> ### Have you searched the existing issues?
> * [x]  This is a unique issue
> 
> ### Describe the bug.
> Is anyone else getting an advertisement on the main screen in the app now? I can easily close it by hitting the X button, but it comes back every time I open the app.
> 
> Example: https://imgur.com/a/n96W4HO

Am about other call, am from Cuba and am restricted, so am wonderin for unrestricted Spotify. Is it possible?

**Comment by APKTre** [comment]
Por favor, perdóname por esta arbitrariedad, pero quiero enviar esto a todos ^^
### Puedes probarlo mientras te relajas después de un día agotador en el trabajo.[REPO Game](https://github.com/R-E-P-O-APKPHAT)
![Screenshot](https://androidphoria.com/wp-content/uploads/R.E.P.O.-para-Android-ya-existe-un-APK-del-juego-696x348.webp)

**Comment by lvvst** [comment]
keeps crashing whenever i open the song

**Comment by testinganything** [comment]
Recently saw an ad in the search tab

Using 5.9.8.
